### PR TITLE
Implement snackbar visibility events

### PR DESCRIPTION
**Pull request by rvasseur31**
Implement events with snackbar visibility.

It's a native feature on android : https://stackoverflow.com/a/33076569

### CODE REVIEW

**Review by cooperka — COMMENTED** [review]
Great code @rvasseur31, thanks for this well-documented enhancement. I will merge this when I can.

Until then, folks can use this right now via your fork as you recommended on another issue -- in package.json:

```
"react-native-snackbar": "rvasseur31/react-native-snackbar#feature/callbacks"
```

**Comment by papidb** [comment]
Hey, @cooperka can you please take a look at this?

File: example/src/App.js [hunk 1]
```diff
@@ -39,7 +52,7 @@ class Example extends Component {
         <TouchableOpacity
           onPress={() => Snackbar.show({
             text: 'Please agree to this.',
-            duration: Snackbar.LENGTH_INDEFINITE,
+            duration: Snackbar.LENGTH_LONG,
```
Comment (cooperka) [inline example/src/App.js]: note to self: this change should be reverted

File: ios/RNSnackBarView.m [hunk 2]
```diff
@@ -37,7 +39,7 @@ @interface RNSnackBarView () {
 @implementation RNSnackBarView
 
 + (void)initialize {
-    DEFAULT_DURATIONS = @{@"-1" : @1500, @"-2" : @2750, @"0" : @INT_MAX};
+    DEFAULT_DURATIONS = @{@"-1" : @1500, @"0" : @2750, @"-2" : @INT_MAX};
```
Comment (cooperka) [inline ios/RNSnackBarView.m:44]: note to self: this changed, not worth reverting

File: yarn.lock [hunk 3]
```diff
@@ -952,160 +952,160 @@
...
+  resolved "https://registry.npmjs.org/@types/istanbul-reports/-/istanbul-reports-1.1.1.tgz"
   integrity sha512-UpYjBi8xefVChsCoBpKShdxTllC9pwISirfoZsUa2AAdQg/Jd2KQGtSbw+ya7GPo7x/wAPlH6JBhKhAsXUEZNA==
   dependencies:
     "@types/istanbul-lib-coverage" "*"
     "@types/istanbul-lib-report" "*"
 
 "@types/mkdirp@^0.5.2":
   version "0.5.2"
-  resolved "https://registry.yarnpkg.com/@types/mkdirp/-/mkdirp-0.5.2.tgz#503aacfe5cc2703d5484326b1b27efa67a339c1f"
+  resolved "https://registry.npmjs.org/@types/mkdirp/-/mkdirp-0.5.2.tgz"
```
Comment (cooperka) [inline yarn.lock]: note to self: revert yarn.lock from npm install

**Comment by cooperka** [comment]
i badly want to keep this library updated, but i don't currently develop react-native applications on a regular basis, so it's a lot of work to keep my dev environment in working order... new laptop, nothing works.

i'm going to ship this code (untested) as a new version; if anything breaks folks can always revert their version & submit a PR to fix it. it's very easy to merge things like that.

thanks again for the contribution, @rvasseur31!

**Comment by cooperka** [comment]
released in v2.7.0 🎉

**Comment by rvasseur31** [comment]
> i badly want to keep this library updated, but i don't currently develop react-native applications on a regular basis, so it's a  lot of work to keep my dev environment in working order... new laptop, nothing works.
>
> i'm going to ship this code (untested) as a new version; if anything breaks folks can always revert their version & submit a PR to fix it. it's very easy to merge things like that.
>
> thanks again for the contribution, @rvasseur31!

Thank you for your time for merging this.

**Pull request merged**